draft-ietf-jsonpath-base.md

```diff
@@ -1011,27 +1012,19 @@ During the iteration process each array element or object member is visited and
 The current item is selected if and only if the boolean expression yields true.
 
 ~~~~ abnf
-boolean-expr     = logical-or-expr
-logical-or-expr  = logical-and-expr *(S "||" S logical-and-expr)
-                                                      ; disjunction
-                                                      ; binds less tightly than conjunction
-logical-and-expr = basic-expr *(S "&&" S basic-expr)  ; conjunction
-                                                      ; binds more tightly than disjunction
-
-basic-expr        = exist-expr /
-                    paren-expr /
-                    relation-expr
-exist-expr        = [logical-not-op S] singular-path  ; path existence or non-existence
-~~~~
-
-Paths in filter expressions are Singular Paths, each of which selects at most one node.
-
-~~~~ abnf
-singular-path     = rel-singular-path / abs-singular-path
-rel-singular-path = "@" *(S (name-segment / index-segment))
-abs-singular-path = root-identifier *(S (name-segment / index-segment))
-name-segment      = "[" name-selector "]" / dot-member-name-shorthand
-index-segment     = "[" index-selector "]"
+boolean-expr           = logical-or-expr
+logical-or-expr        = logical-and-expr *(S "||" S logical-and-expr)
+                                                            ; disjunction
+                                                            ; binds less tightly than conjunction
+logical-and-expr       = basic-expr *(S "&&" S basic-expr)  ; conjunction
+                                                            ; binds more tightly than disjunction
+
+basic-expr              = exist-expr /
```

Comment:
> I would group this paren-expr, relation-ex, exist-expr.

Ok. The current order of `basic-expr` matches the order in which the alternatives are specified lower down. Feel free to regroup this later. 

> 
> The ABNF here conflicts with another PR already merged, but this particular change looks innocuous otherwise.

The delta clash was a little painful. It is now resolved and I hope I haven't lost any pieces in the process!
> 
> The cutting down to 69 columns needs to be done at some point; maybe after this merge will be a good start.

Eek! Ouch! That's very narrow. Is someone wanting to print this spec on punched cards?

Reply:
While RFC 2223 is obsoleted by RFC 7322, we still follow its Section 3a, which says:

```
      Each line must be limited to 72 characters followed by carriage
      return and line feed.
```

With the usual three characters indentation, this leaves us 69 characters.